server/storage_s3.go

```diff
@@ -141,6 +148,13 @@ func createS3Integration(keys map[string]string, properties map[string]bool) (mo
 	if err != nil {
 		return models.Integration{}, err
 	} else if !exist {
+		if value, ok := keys["secret_key"]; ok {
+			encryptedValue, err := EncryptAES([]byte(value))
+			if err != nil {
+				return models.Integration{}, err
+			}
+			keys["secret_key"] = encryptedValue
```

Comment:
this is not good, you are set it here then keys go into the integrationToUpdate var and then this is broadcast to the clients with an encrypted value